src/mqt/predictor/utils.py

```diff
@@ -21,10 +24,16 @@ def timeout_handler(signum, frame):  # Custom signal handler
     try:
         res = func(*args)
     except TimeoutException:
-        print("Calculation/Generation exceeded timeout limit for ", func, args[1:])
+        logger.debug(
+            "Calculation/Generation exceeded timeout limit for "
+            + ", "
+            + func
+            + ", "
+            + args[1:]
+        )
         return False
     except Exception as e:
-        print("Something else went wrong: ", e)
+        logger.error("Something else went wrong: " + ", " + e)
         return False
```

Comment:
should probably re-raise the exception

Reply:
In this case I like to stick to only logging the error. During the compilation routines for all necessary compilation paths, I remember that sometimes weird Exceptions were raised for certain number of qubits from within qiskit. I cannot really recall anymore what those instances were, but a exception would terminate the whole compilation routine for alle paths which would maybe be disadvantageous.

Reply:
Fine with me. As long as there is a reason for it, logging errors and continuing is fair enough.